src/server/authentication.js

```diff
@@ -0,0 +1,114 @@
+const argon2 = require('argon2');
+const validator = require('validator');
+const jwt = require('jsonwebtoken');
+
+module.exports = psql => ({
+
+  /*
+      first checks if username/email already exist
+      returns true if successful
+     */
+  register: async (user) => {
+    const { username } = user;
+    const { email } = user;
+    const { password } = user;
+    let checkUsers;
+    let hash;
+
+    if (!validator.isEmail(email)) {
+      return false;
+    }
+
+    try {
+      checkUsers = await psql('users')
+        .where('username', username)
+        .orWhere('email', email)
+        .returning('user_id');
+
+      if (checkUsers.length > 0) {
+        return 'username or email already taken';
+      }
+
+      hash = await argon2.hash(password);
+      await psql('users').insert({
+        username,
+        email,
+        password: hash,
+        date_created: new Date(),
+      }).returning('username');
+
+      return true;
+    } catch (e) {
+      console.log(e);
+      return 'failed to create user';
```

Comment:
If this catch block is for unexpected "really-shouldn't-happen" errors, let the user know it's not their falt
```suggestion
      return 'unexpected error during registration';
```

If this catch is for a known error that a library function throws, wrap the try-catch tightly around that function, and give a specific error message